cmd/server/main.go

```diff
@@ -1,53 +1,169 @@
 package main
 
 import (
+	"context"
+	"flag"
 	"fmt"
-	"net"
-	"time"
+	"log"
+	"runtime"
 
+	goGrpc "google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rpc"
 	"github.com/onflow/flow-evm-gateway/api"
+	"github.com/onflow/flow-evm-gateway/storage"
+	"github.com/onflow/flow-go-sdk"
+	"github.com/onflow/flow-go-sdk/access/grpc"
+	"github.com/rs/zerolog"
+)
+
+const (
+	accessURL    = "access-001.devnet49.nodes.onflow.org:9000"
+	coinbaseAddr = "0xf02c1c8e6114b1dbe8937a39260b5b0a374432bb"
 )
 
 func main() {
-	server := api.NewRPCServer()
-	defer server.Stop()
+	var network, coinbase string
+
+	flag.StringVar(&network, "network", "testnet", "network to connect the gateway to")
+	flag.StringVar(&coinbase, "coinbase", coinbaseAddr, "coinbase address to use for fee collection")
+	flag.Parse()
+
+	config := &api.Config{}
+	config.Coinbase = common.HexToAddress(coinbase)
+	if network == "testnet" {
+		config.ChainID = api.FlowEVMTestnetChainID
+	} else if network == "mainnet" {
+		config.ChainID = api.FlowEVMMainnetChainID
+	} else {
+		panic(fmt.Errorf("unknown network: %s", network))
+	}
+
+	store := storage.NewStore()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
-	listener, err := net.Listen("tcp", "127.0.0.1:0")
+	runServer(config, store)
+	runIndexer(ctx, store)
+
+	runtime.Goexit()
+}
+
+func runIndexer(ctx context.Context, store *storage.Store) {
+	flowClient, err := grpc.NewBaseClient(
+		accessURL,
+		goGrpc.WithTransportCredentials(insecure.NewCredentials()),
+	)
 	if err != nil {
-		fmt.Println("can't listen:", err)
+		panic(err)
 	}
-	defer listener.Close()
-	go server.ServeListener(listener)
-
-	requests := []string{
-		`{"jsonrpc":"2.0","id":1,"method":"eth_blockNumber","params": []}`,
-		`{"jsonrpc":"2.0","id":2,"method":"eth_chainId","params": []}`,
-		`{"jsonrpc":"2.0","id":3,"method":"eth_syncing","params": []}`,
-		`{"jsonrpc":"2.0","id":4,"method":"eth_getBalance","params": ["0x407d73d8a49eeb85d32cf465507dd71d507100c1"]}`,
-		`{"jsonrpc":"2.0","id":5,"method":"eth_getBlockTransactionCountByNumber","params": ["0x4E4ee"]}`,
+
+	latestBlockHeader, err := flowClient.GetLatestBlockHeader(ctx, true)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Printf("Latest Block Height: %d\n", latestBlockHeader.Height)
```

Comment:
I suggest you add a logging library and replace fmt outputs.

Reply:
Updated in https://github.com/onflow/flow-evm-gateway/pull/11/commits/f55dc7b362b529c43f36825694bcf9221282bb12